k9mail/src/main/java/com/fsck/k9/crypto/MessageDecryptVerifier.java

```diff
@@ -194,16 +194,22 @@ private static Part findPrimaryPartInAlternative(Part part) {
         return inlineParts;
     }
 
+    /**
+     * @param part potentially with signature data.
+     * @return may be null if no valid signature data found.
+     */
     public static byte[] getSignatureData(Part part) throws IOException, MessagingException {
         if (isPartMultipartSigned(part)) {
             Body body = part.getBody();
             if (body instanceof Multipart) {
                 Multipart multi = (Multipart) body;
-                BodyPart signatureBody = multi.getBodyPart(1);
-                if (isSameMimeType(signatureBody.getMimeType(), APPLICATION_PGP_SIGNATURE)) {
-                    ByteArrayOutputStream bos = new ByteArrayOutputStream();
-                    signatureBody.getBody().writeTo(bos);
-                    return bos.toByteArray();
+                if (multi.getCount() > 1) {//Bug #2225 mailing list strips part out
```

Comment:
this comment isn't really going to be helpful some time from now, please drop it